llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -4600,6 +4651,9 @@ static Value *simplifySelectWithICmpEq(Value *CmpLHS, Value *CmpRHS,
                              /* DropFlags */ nullptr, MaxRecurse) == FalseVal)
     return FalseVal;
 
+  if (simplifySpecificBitOp(CmpLHS, CmpRHS, TrueVal, FalseVal))
+    return FalseVal;
```

Comment:
```suggestion
  if (Value *V = simplifySpecificBitOp(CmpLHS, CmpRHS, TrueVal, FalseVal))
    return V;
```
It is unsafe since we may handle more patterns in the future...

Reply:
got it!